src/main/java/data/Parser.java

```diff
@@ -0,0 +1,242 @@
+package data;
+
+import command.*;
+import task.Deadline;
+import task.Event;
+import task.ToDo;
+
+import java.time.LocalDate;
+import java.time.LocalDateTime;
+import java.time.YearMonth;
+import java.time.format.DateTimeFormatter;
+import java.time.format.DateTimeParseException;
+
+/**
+ * Class that parses a user's input.
+ */
+public class Parser {
+    /** The different formats of date/time allowed as input */
+    private final static String DATE_AND_TIME_FORMAT[] = {
+        "M/y", "d/M/y", "d-M-y", "M/y HHmm", "d/M/y HHmm", "d-M-y HHmm"
+    };
+
+    /**
+     * Checks the keyword of the input and return the corresponding command.
+     *
+     * @param userInput The text that the user inputs
+     * @return A Command that does what the keyword intended when executed
+     */
+    public static Command parse(String userInput) {
+        String[] userInputArray = userInput.split(" ");
+        String keyword = userInputArray[0];
+
+        switch (keyword) {
+
+        case "deadline":
+            return prepareAddCommand("deadline", userInput, userInputArray, "/by");
+
+        case "event":
+            return prepareAddCommand("event", userInput, userInputArray, "/at");
+
+        case "todo":
+            return prepareAddCommand("todo", userInput, userInputArray, "");
+
+        case "done":
+            return prepareDoneCommand(userInputArray);
+
+        case "list":
+            return prepareListCommand(userInputArray);
+
+        case "delete":
+            return prepareDeleteCommand(userInputArray);
+
+        case "bye":
+            return prepareExitCommand(userInputArray);
+
+        case "find":
+            return prepareFindCommand(userInputArray);
+
+        default:
+            throw new DukeException("Invalid Keyword.");
+        }
```

Comment:
I really like how you separated the commands into different methods, this makes your code very neat and concise :)))